### PR TITLE
Fix a number of lint perf and safety issues in torch

### PR DESCRIPTION
**Pull request by r-barnes**
Differential Revision: D29037012

### CODE REVIEW

File: torch/csrc/utils/python_arg_parser.h [hunk 1]
```diff
@@ -384,30 +384,30 @@ inline std::vector<int64_t> PythonArgs::intlist(int i) {
...
   PyObject* arg = args[i];
-  auto size = signature.params[i].size;
-  if (size > 0 && THPUtils_checkLong(arg)) {
-    return std::vector<int64_t>(size, THPUtils_unpackIndex(arg));
+  const auto size1 = signature.params[i].size;
+  if (size1 > 0 && THPUtils_checkLong(arg)) {
+    return std::vector<int64_t>(size1, THPUtils_unpackIndex(arg));
   }
   auto tuple = PyTuple_Check(arg);
   // NOLINTNEXTLINE(bugprone-branch-clone)
```
Comment (albanD) [inline torch/csrc/utils/python_arg_parser.h:392]: Should the comment be removed here?

File: torch/csrc/jit/tensorexpr/ir_simplifier.cpp [hunk 2]
```diff
@@ -1921,15 +1921,15 @@ const Expr* simplifyRoundModPattern(const Polynomial* poly) {
...
     // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
-    for (int i = mods.size() - 1; i >= 0; i--) {
+    for (int64_t i = mods.size() - 1; i >= 0; i--) {
       const Term* m = mods[i];
       const Mod* mod = dynamic_cast<const Mod*>(m->variables()[0]);
       CHECK(mod);
       const Expr* mod_lhs = IRSimplifier::simplify(mod->lhs());
       const Expr* mod_rhs = IRSimplifier::simplify(mod->rhs());
       bool merged = false;
       // NOLINTNEXTLINE(bugprone-narrowing-conversions,cppcoreguidelines-narrowing-conversions)
```
Comment (albanD) [inline torch/csrc/jit/tensorexpr/ir_simplifier.cpp:1931]: Comments can be removed in multiple places in this file no?

**Review by albanD — APPROVED** [review]
Looks good to me beyond a couple comments to remove.
Thanks for taking the time to do this!

File: torch/csrc/jit/tensorexpr/ir_simplifier.cpp (hunk 2 above)
Comment (r-barnes) [reply torch/csrc/jit/tensorexpr/ir_simplifier.cpp:1931]: This still makes a conversion from `size_t` to `int64_t`, which is narrowing. It's just not as bad now.

**Pull request closed without merging**